src/metabase/search/core.clj

```diff
@@ -76,3 +76,10 @@
                             seq)]
       ;; We need to delay execution to handle deletes, which alert us *before* updating the database.
       (search.ingestion/ingest-maybe-async! updates))))
+
+(defn delete!
+  "Given a model and a list of model's ids, remove corresponding search entries."
+  [model ids]
+  (doseq [e                      (search.engine/active-engines)
+          {:keys [search-model]} (get (search.spec/model-hooks) model)]
```

Comment:
This is not quite right, it will include all the down-stream models, which will not have the same ids. I think it's best not to worry about transitive deletes for now, which will need more machinery, and instead only handle the search models whose specification have the given `:model`.

Something like:

```clojure
(doseq [search-model (map :name (filter (comp #{model} :model) (vals (search.spec/specifications))))] ...)
```